### ISSUE TITLE
Please support "DEFERRABLE INITIALLY DEFERRED"

### ISSUE THREAD

**Issue by yodhcn** [knex/knex#4292]
# Feature request
Is there any chance to have a Knex native way to set "DEFERRABLE INITIALLY DEFERRED" on a table with foreign key constraints ?

```
knex.schema.table('users', function (table) {
  table.integer('user_id').unsigned()
  table.foreign('user_id').references('Items.user_id_in_items').deferred().
})
```

**Comment by kibertoad** [comment]
@yodhcn PR for this would be most welcome, and if you need any help with finding the right spot in source or writing tests, would be more than happy to help.

**Comment by Alucard17** [comment]
hey @kibertoad  can I take this issue?

**Comment by Alucard17** [comment]
hey @yodhcn can you provide the compiled SQL query of above mentioned example.

**Comment by kibertoad** [comment]
Released in 0.95.8
